controller/user.go

```diff
@@ -68,3 +72,55 @@ func (c *UserController) Show(ctx *app.ShowUserContext) error {
 		return ctx.OK(ConvertToAppUser(ctx.RequestData, user, identity, true))
 	})
 }
+
+// ListResources returns a list of resources in which the current user has a role
+func (c *UserController) ListResources(ctx *app.ListResourcesUserContext) error {
+	// retrieve the user's identity ID from the token
+	identityID, err := c.tokenManager.Locate(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "Bad Token")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("bad or missing token"))
+	}
+	// check the requested resource type
+	if ctx.Type == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterErrorFromString("type", "", "missing 'type' query parameter in the request"))
+	}
+	var resourceType string
+	switch *ctx.Type {
+	case "spaces":
+		resourceType = authorization.ResourceTypeSpace
+	}
+	if resourceType == "" {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("type", *ctx.Type))
```

Comment:
how about giving supported parameters in error to help cosumer of this api?

Reply:
ok, let me change the error to include the list of supported types (ie, `spaces` only for now)